### ISSUE TITLE
BUG: GeoDataFrame.to_parquet writes geometry in EWKB instead of ISO WKB required by spec

### ISSUE THREAD

**Issue by himikof** [geopandas/geopandas#2654]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [x] (optional) I have confirmed this bug exists on the main branch of geopandas.

---

#### Code Sample, a copy-pastable example

```python
import geopandas
geopandas.GeoDataFrame(geometry=geopandas.GeoSeries.from_wkt(['POINT Z (1 2 3)'])) \
    .to_parquet('pointz.parquet', schema_version='0.4.0')
import pyarrow.parquet as pq
print(pq.read_table('pointz.parquet').to_pydict())
{'geometry': [b'\x01\x01\x00\x00\x80\x00\x00\x00\x00\x00\x00\xf0?\x00\x00\x00\x00\x00\x00\x00@\x00\x00\x00\x00\x00\x00\x08@']}

```

#### Problem description

The [0.4.0 GeoParquet specification](https://github.com/opengeospatial/geoparquet/blob/v0.4.0/format-specs/geoparquet.md#encoding) explicitly requires encoding extended geometry types in ISO WKB:
> using codes for 3D geometry types in the [1001,1007] range

The geometry was instead encoded using EWKB 0x80000001 code, making the resulting file invalid GeoParquet.
As I understand, GEOS has `GEOSWKBWriter_setFlavor` since 3.10 that allows to write ISO WKB correctly. I understand that there are currently several different code paths implementing WKB writing, and implementing such support for them all could be challenging. But even raising an error early on non-supported geometry types would be in my opinion preferable to silently writing invalid data files.

#### Expected Output
`{'geometry': [b'\x01\xe9\x03\x00\x00\x00\x00\x00\x00\x00\x00\xf0?\x00\x00\x00\x00\x00\x00\x00@\x00\x00\x00\x00\x00\x00\x08@']}`
Alternatively, an error explaining that serializing POINT Z geometry type in GeoParquet is not currently supported.

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.9.9 (main, Nov 19 2021, 00:00:00)  [GCC 10.3.1 20210422 (Red Hat 10.3.1-1)]
executable : /home/nofitserov/.cache/pypoetry/virtualenvs/test-d9F45URs-py3.9/bin/python
machine    : Linux-5.14.18-100.fc33.x86_64-x86_64-with-glibc2.32

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.8.1
GEOS lib   : /usr/lib64/libgeos_c.so
GDAL       : 3.4.3
GDAL data dir: /home/nofitserov/.cache/pypoetry/virtualenvs/test-d9F45URs-py3.9/lib64/python3.9/site-packages/fiona/gdal_data
PROJ       : 9.1.0
PROJ data dir: /home/nofitserov/.cache/pypoetry/virtualenvs/test-d9F45URs-py3.9/lib64/python3.9/site-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.12.1
numpy      : 1.23.5
pandas     : 1.5.1
pyproj     : 3.4.0
shapely    : 1.8.5.post1
fiona      : 1.8.22
geoalchemy2: None
geopy      : None
matplotlib : 3.6.2
mapclassify: 2.4.3
pygeos     : None
pyogrio    : v0.4.2
psycopg2   : None
pyarrow    : 10.0.0
rtree      : None

</details>

**Comment by martinfleis** [comment]
Thanks. Support for this has been added in https://github.com/shapely/shapely/pull/1628 and just needs to be used here now.

**Comment by jorisvandenbossche** [comment]
Yes, I intentionally included that last-minute in shapely 2.0 so we could already start using it here. I was planning to do a PR for this (after https://github.com/geopandas/geopandas/pull/2663), but feel free to beat me to it.

**Comment by jorisvandenbossche** [comment]
I opened a PR for this -> https://github.com/geopandas/geopandas/pull/2760

Open question is whether we want to warn for non-conformant files or not in case the user doesn't have Shapely >=2.0 installed?
